From a3b768379810bd34f22c97ccb01c81f8d866401f Mon Sep 17 00:00:00 2001
From: Wang Zaikuo <zaikuo.wang@intel.com>
Date: Thu, 16 Aug 2018 15:38:14 +0800
Subject: [PATCH 2254/2313] media: intel-ipu4: isys: be_soc only sets route 0
 immutable

Message for Open Source:
Only set BE SOC route 0 immutable to allow user to reconfigure
other routes at runtime

Message for Internal:
[Issue/Feature] ti964 case could not work without this fix
[Root Cause/Changes] ti964 has a regression caused by patch: 345645,
because all of routes of be_soc flags included IMMUTABLE, so the
stream infomation could not set at runtime;

Change-Id: Ie3e1b856d161ce6a1d04d9444660325df33d3355
Signed-off-by: Wang Zaikuo <zaikuo.wang@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>

Conflicts:
	drivers/media/pci/intel/ipu-isys-csi2-be-soc.c
---
 drivers/media/pci/intel/ipu-isys-csi2-be-soc.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-isys-csi2-be-soc.c b/drivers/media/pci/intel/ipu-isys-csi2-be-soc.c
index 8648b5f..4ee7a6d 100644
--- a/drivers/media/pci/intel/ipu-isys-csi2-be-soc.c
+++ b/drivers/media/pci/intel/ipu-isys-csi2-be-soc.c
@@ -284,13 +284,14 @@ int ipu_isys_csi2_be_soc_init(struct ipu_isys_csi2_be_soc *csi2_be_soc,
 		    = 0;
 	}
 	for (i = 0; i < NR_OF_CSI2_BE_SOC_STREAMS; i++) {
-		csi2_be_soc->asd.route[i].flags = V4L2_SUBDEV_ROUTE_FL_ACTIVE |
-		    V4L2_SUBDEV_ROUTE_FL_IMMUTABLE;
+		csi2_be_soc->asd.route[i].flags = V4L2_SUBDEV_ROUTE_FL_ACTIVE;
 		bitmap_set(csi2_be_soc->asd.stream[CSI2_BE_SOC_PAD_SINK(i)].
 			   streams_stat, 0, 1);
 		bitmap_set(csi2_be_soc->asd.stream[CSI2_BE_SOC_PAD_SOURCE(i)].
 			   streams_stat, 0, 1);
 	}
+	csi2_be_soc->asd.route[0].flags |= V4L2_SUBDEV_ROUTE_FL_IMMUTABLE;
+#endif
 	mutex_unlock(&csi2_be_soc->asd.mutex);
 	for (i = 0; i < NR_OF_CSI2_BE_SOC_SOURCE_PADS; i++) {
 		snprintf(csi2_be_soc->av[i].vdev.name,
-- 
2.7.4

